lib/workers/repository/process/index.ts

```diff
@@ -130,6 +130,16 @@ export async function extractDependencies(
     const extracted: Record<string, Record<string, PackageFile[]>> = {};
     for (const baseBranch of config.baseBranches) {
       addMeta({ baseBranch });
+
+      // TODO: initialize these two variables
+      const isForkMode = true;
+      const isGitHub = true;
```

Comment:
How to determine if we are in forkMode and that the scm in use in in fact git(hub)?

I think that if we are in forkMode but platform is non-git, we should log an error and abort since we cannot bring fork up to date. Perhaps this will happen automatically for scm implementations that do not implement `syncForkBranch()`?

Reply:
So I added a function `scm.hasUpstream()` as a means to determine if we are in forkingMode. It would be easier if we could just read some boolean from `config` object, but I looked but `config.isFork()` seems to indicate whether a discovered repo is a fork and should be skipped, while `config.forkProcessing` seems to be mostly for disabling fork processing. Also I cannot find the `upstreamUrl` variable used during git init from `LocalRepoConfig` does not seem to be available from the process worker. Anything I missed?